src/ripple/rpc/tests/AccountSet.test.cpp

```diff
@@ -0,0 +1,309 @@
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/protocol/JsonFields.h>
+#include <ripple/protocol/Feature.h>
+#include <ripple/test/jtx.h>
+#include <ripple/basics/StringUtilities.h>
+
+namespace ripple {
+namespace test {
+
+class AccountSet_test : public beast::unit_test::suite
+{
+public:
+
+    void testNullAccountSet()
+    {
+        using namespace jtx;
+        Env env(*this);
+        Account const alice ("alice");
+        env.fund(XRP(10000), noripple(alice));
+        //ask for the ledger entry - account root, to check its flags
+        auto const jrr = env.le(alice);
+        BEAST_EXPECT(env.le(alice)->getFieldU32(sfFlags) == 0u);
+    }
+
+    void testSetAndReset(unsigned int flag_val, std::string const& label)
+    {
+        using namespace jtx;
+        Env env(*this);
+        Account const alice ("alice");
+        env.fund(XRP(10000), noripple(alice));
+        env.memoize("eric");
+        env(regkey(alice, "eric"));
+
+        log << "Set/Reset for " << label << std::endl;
+        unsigned int orig_flags = env.le(alice)->getFieldU32(sfFlags);
+
+        env.require(nflags(alice, flag_val));
+        env(fset(alice, flag_val), sig(alice));
+        env.require(flags(alice, flag_val));
+        env(fclear(alice, flag_val));
+        env.require(nflags(alice, flag_val));
+        uint32 now_flags = env.le(alice)->getFieldU32(sfFlags);
+        BEAST_EXPECT(now_flags == orig_flags);
+    }
+
+    void testSetAndResetAccountTxnID()
+    {
+        using namespace jtx;
+        Env env(*this);
+        Account const alice ("alice");
+        env.fund(XRP(10000), noripple(alice));
+
+        log << "Set/Reset for AccountTxnID" << std::endl;
+        unsigned int orig_flags = env.le(alice)->getFieldU32(sfFlags);
+
+        // asfAccountTxnID is special and not actually set as a flag,
+        // so we check the field presence instead
+        BEAST_EXPECT(! env.le(alice)->isFieldPresent(sfAccountTxnID));
+        env(fset(alice, asfAccountTxnID), sig(alice));
+        BEAST_EXPECT(env.le(alice)->isFieldPresent(sfAccountTxnID));
+        env(fclear(alice, asfAccountTxnID));
+        BEAST_EXPECT(! env.le(alice)->isFieldPresent(sfAccountTxnID));
+        uint32 now_flags = env.le(alice)->getFieldU32(sfFlags);
+        BEAST_EXPECT(now_flags == orig_flags);
+    }
+
+    void testSetNoFreeze()
+    {
+        using namespace jtx;
+        Env env(*this);
+        Account const alice ("alice");
+        env.fund(XRP(10000), noripple(alice));
+        env.memoize("eric");
+        env(regkey(alice, "eric"));
+
+        log << "Set NoFreeze -- cannot be unset" << std::endl;
```

Comment:
`std::endl` is not needed, `log` flushes after every write:
https://github.com/ripple/rippled/blob/develop/src/beast/extras/beast/unit_test/suite.hpp#L67